imqsauth/http.go

```diff
@@ -249,33 +250,74 @@ func (x *ImqsCentral) ResetPasswordStart(userId authaus.UserId, isNewAccount boo
-		return http.StatusForbidden, "Error resetting password: " + err.Error()
+		return http.StatusBadRequest, "Error retrieving auth user type: " + err.Error()
 	}
 
-	// &welcome=true -- use this for welcome email
-	strUserId := strconv.FormatInt(int64(userId), 10)
-	identity, errGetIdentity := x.Central.GetIdentityFromUserId(userId)
-	if errGetIdentity != nil {
-		return http.StatusBadRequest, "Unrecognized userid: " + err.Error()
+	var mailQuery string
+	var mailBody string
+	if user.Type.CanSetPassword() {
+		token, err := x.Central.ResetPasswordStart(userId, time.Now().Add(time.Duration(expireSeconds)*time.Second))
+		if err != nil {
+			return http.StatusForbidden, "Error resetting password: " + err.Error()
+		}
+
+		mailQuery, err = x.createDefaultMailQuery(user, token, isNewAccount, expireSeconds)
+		if err != nil {
+			return http.StatusServiceUnavailable, "Error constructing reset URL: " + err.Error()
+		}
+	} else if user.Type == authaus.UserTypeLDAP {
+		mailQuery, mailBody = x.createLDAPMailQueryAndBody(user)
 	}
-	resetUrl, err := x.makeAbsoluteUrl("/#resetpassword=true&identity=" + url.QueryEscape(identity) + "&userid=" + url.QueryEscape(strUserId) + "&token=" + url.QueryEscape(token))
+
+	return x.buildMailRequestAndSend(mailQuery, mailBody)
+}
+
+func (x *ImqsCentral) createDefaultMailQuery(user authaus.AuthUser, token string, isNewAccount bool, expireSeconds float64) (string, error) {
+	strUserId := strconv.FormatInt(int64(user.UserId), 10)
+	resetUrl, err := x.makeAbsoluteUrl("/#resetpassword=true&identity=" + url.QueryEscape(user.Email) + "&userid=" + url.QueryEscape(strUserId) + "&token=" + url.QueryEscape(token))
 	if err != nil {
-		return http.StatusServiceUnavailable, "Error constructing reset URL: " + err.Error()
+		return "", err
 	}
 	if isNewAccount {
 		resetUrl += "&welcome=true"
 	}
 
-	mailQuery := fmt.Sprintf("email=%v&resetUrl=%v&expireTime=%.0f", url.QueryEscape(identity), url.QueryEscape(resetUrl), expireSeconds)
+	mailQuery := "passwordReset?"
+	mailQuery += fmt.Sprintf("email=%v&resetUrl=%v&expireTime=%.0f", url.QueryEscape(user.Email), url.QueryEscape(resetUrl), expireSeconds)
 	if isNewAccount {
 		mailQuery += "&newAccount=true"
 	} else {
 		mailQuery += "&newAccount=false"
 	}
 
-	sendMailReq, err := http.NewRequest("POST", "https://imqs-mailer.appspot.com/passwordReset?"+mailQuery, nil)
+	return mailQuery, nil
+}
+
+func (x *ImqsCentral) createLDAPMailQueryAndBody(user authaus.AuthUser) (string, string) {
+	var mailQuery string
+	var mailBody string
+	mailQuery += "sendEmail?"
+	mailQuery += fmt.Sprintf("emailTo=%v&subject=%v&ishtml=True", url.QueryEscape(user.Email), url.QueryEscape("IMQS Reset Password"))
+	if len(x.Config.Authaus.LDAP.SysAdminEmail) > 0 {
+		mailBody += fmt.Sprintf("This is an automated response.<br><br>To reset your password, please contact your System Administrator (%v).<br><br> Note* Please use your system username and password.", x.Config.Authaus.LDAP.SysAdminEmail)
```

Comment:
I'm not sure that the words "Please use your system username and password" add any value. I wouldn't know what that means. What is the "system" here?